File: benchmarks/cases/coprime/coprime.dart
// Count coprime pairs (i,j), 1<=i,j<=N — nested iteration + Euclidean gcd.

int gcd(int a, int b) => b == 0 ? a : gcd(b, a % b);

void main() {
  const n = 2000;
  int acc = 0;
  for (int i = n; i > 0; i--) {
    for (int j = n; j > 0; j--) {
      if (gcd(i, j) == 1) {
        acc++;
      }
    }
  }
  print(acc);
}

File: benchmarks/cases/coprime/coprime.cs
// Count coprime pairs (i,j), 1<=i,j<=N — nested iteration + Euclidean gcd.

long n = 2000;
long acc = 0;
for (long i = n; i > 0; i--)
{
    for (long j = n; j > 0; j--)
    {
        if (Gcd(i, j) == 1) acc++;
    }
}
System.Console.WriteLine(acc);

static long Gcd(long a, long b) => b == 0 ? a : Gcd(b, a % b);
